base/reduce.jl

```diff
@@ -703,22 +703,3 @@ function count(pred, a::AbstractArray)
     return n
 end
 count(itr) = count(identity, itr)
-
-"""
-    countnz(A) -> Integer
-
-Counts the number of nonzero values in array `A` (dense or sparse). Note that this is not a constant-time operation.
-For sparse matrices, one should usually use [`nnz`](@ref), which returns the number of stored values.
-
-```jldoctest
-julia> A = [1 2 4; 0 0 1; 1 1 0]
-3×3 Array{Int64,2}:
- 1  2  4
- 0  0  1
- 1  1  0
-
-julia> countnz(A)
-6
-```
-"""
-countnz(a) = count(x -> x != 0, a)
```

Comment:
Note that the deprecation is different from the current definition, which (as you remarked) is technically breaking. I think the best approach would be to use `Base.depwarn` to print a custom deprecation warning proposing both forms, but continue calling `x -> x != 0` under the hood.

Reply:
I've changed the deprecation to suggest both versions following your suggestion.